app/Http/Controllers/WebAuthn/WebAuthnLoginController.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Http\Controllers\WebAuthn;
+
+use App\Exceptions\UnauthenticatedException;
+use App\Models\User;
+use App\Pipelines\AssertionValidator;
+use Illuminate\Contracts\Support\Responsable;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Auth;
+use Laragear\WebAuthn\Assertion\Validator\AssertionValidation;
+use Laragear\WebAuthn\Http\Requests\AssertedRequest;
+use Laragear\WebAuthn\Http\Requests\AssertionRequest;
+use function response;
+
+class WebAuthnLoginController
+{
+	/**
+	 * Returns the challenge to assertion.
+	 *
+	 * @param AssertionRequest $request
+	 *
+	 * @return \Illuminate\Contracts\Support\Responsable
+	 */
+	public function options(AssertionRequest $request): Responsable
+	{
+		return $request->toVerify($request->validate(['user_id' => 'sometimes|int'])['user_id']);
+	}
+
+	/**
+	 * Log the user in.
+	 *
+	 * @param AssertedRequest $request
+	 *
+	 * @return \Illuminate\Http\Response
+	 */
+	public function login(AssertedRequest $request, AssertionValidator $assertion): Response
```

Comment:
If I am not mistaken, we either throw an exception or return an empty `204` response. In this case it also OK, to not return anything which is the same as returning `204` to the client.

```suggestion
	public function login(AssertedRequest $request, AssertionValidator $assertion): void
```

Reply:
fixed [773f085](https://github.com/LycheeOrg/Lychee/pull/1469/commits/773f0855084f1c229cfe17229574fd747be83d02)